packages/flutter_tools/lib/src/windows/windows_version_validator.dart

```diff
@@ -108,6 +108,11 @@ class ProcessLister {
 
   Future<ProcessResult> getProcessesWithPath() async {
     const String argument = 'Get-Process | Format-List Path';
-    return processManager.run(<String>['powershell', '-command', argument]);
+    const List<String> psArgs = <String>['-command', argument];
+    try {
+      return await processManager.run(<String>['powershell', ...psArgs]);
+    } on ProcessPackageExecutableNotFoundException {
+      return processManager.run(<String>['pwsh', ...psArgs]);
+    }
```

Comment:
This LGTM as-is, but I think there is a world where we can detect that powershell is unavailable (via `ProcessManager.canRun`) and then make the validator fail with a `ValidationResult.missing` that has a more specific message rather than throwing an exception on relying on higher-level error handling. (This is just an FYI; my hope is that this knowledge is useful for future work on validators).

However, this kind of situation is unlikely to happen again very often, and the error message generated in that case is still sufficient to figure out what's going wrong here—hence the LGTM.

Reply:
That's a great point! I definitely just skipped past all the validation logic at the top of the file... :)

I've gone ahead and added a validation step and made more use of `ProcessManager.canRun` rather than attempting to run executables to determine whether or not they exist.